src/puya/awst/nodes.py

```diff
@@ -322,6 +418,11 @@ def accept(self, visitor: ExpressionVisitor[T]) -> T:
 
 @enum.unique
 class BytesEncoding(enum.StrEnum):
+    """
+    Captures the source encoding of a bytes constant so that an appropriate encoding
+    can be used to produce a human-readable value in code output.
```

Comment:
it's more about preserving constant in it's original form/encoding, where possible.